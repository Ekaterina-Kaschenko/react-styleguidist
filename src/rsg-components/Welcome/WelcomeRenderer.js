import React, { PropTypes } from 'react';
import Markdown from 'rsg-components/Markdown';
import Styled from 'rsg-components/Styled';
import { DOCS_COMPONENTS, DOCS_DOCUMENTING } from '../../../scripts/consts';

/* eslint-disable max-len */

const styles = () => ({
	root: {
		maxWidth: 900,
		margin: [[0, 'auto']],
		padding: 30,
	},
});

export function WelcomeRenderer({ classes, components, examples }) {
	return (
		<div className={classes.root}>
			<Markdown text={'# Welcome to React Styleguidist!'} />
			{components && (
				<Markdown
					text={`
## Point Styleguidist to your React components

We couldn’t find any components defined in your style guide config or using the default pattern \`src/components/**/*.js\`.

Create **styleguide.config.js** file in your project root directory like this:

    module.exports = {
      components: 'src/components/**/*.js'
    };

Read more in the [locating components guide](${DOCS_COMPONENTS}).
					`}
				/>
			)}
			{examples && (
				<Markdown
					text={`

## Write your first example

Create **Readme.md** or **ComponentName.md** file in the component’s folder like this:

    React component example:

        &lt;Button size="large"&gt;Push Me&lt;/Button&gt;

Read more in the [documenting components guide](${DOCS_DOCUMENTING}).
					`}
				/>
			)}
		</div>
	);
}

WelcomeRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	components: PropTypes.bool,
	examples: PropTypes.bool,
};

export default Styled(styles)(WelcomeRenderer);
