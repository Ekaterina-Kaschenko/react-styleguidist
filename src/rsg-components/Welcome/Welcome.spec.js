import React from 'react';
import { WelcomeRenderer } from './WelcomeRenderer';

it('renderer should welcome screen with no components message', () => {
	const actual = shallow(
		<WelcomeRenderer classes={{}} components />
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should welcome screen with no examples message', () => {
	const actual = shallow(
		<WelcomeRenderer classes={{}} examples />
	);

	expect(actual).toMatchSnapshot();
});
