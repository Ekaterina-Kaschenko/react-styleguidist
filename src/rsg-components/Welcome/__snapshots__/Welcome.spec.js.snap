exports[`test renderer should welcome screen with no components message 1`] = `
<div>
  <_class
    text="# Welcome to React Styleguidist!" />
  <_class
    text="
  ## Point Styleguidist to your React components
  
  We couldn’t find any components defined in your style guide config or using the default pattern \`src/components/**/*.js\`.
  
  Create **styleguide.config.js** file in your project root directory like this:
  
      module.exports = {
        components: \'src/components/**/*.js\'
      };
  
  Read more in the [locating components guide](https://github.com/styleguidist/react-styleguidist/blob/master/docs/Components.md).
  					" />
</div>
`;

exports[`test renderer should welcome screen with no examples message 1`] = `
<div>
  <_class
    text="# Welcome to React Styleguidist!" />
  <_class
    text="
  
  ## Write your first example
  
  Create **Readme.md** or **ComponentName.md** file in the component’s folder like this:
  
      React component example:
  
          &lt;Button size=\"large\"&gt;Push Me&lt;/Button&gt;
  
  Read more in the [documenting components guide](https://github.com/styleguidist/react-styleguidist/blob/master/docs/Documenting.md).
  					" />
</div>
`;
